snuba/datasets/groups.py

```diff
@@ -0,0 +1,119 @@
+import re
+
+from datetime import timedelta
+from typing import Mapping, Sequence
+
+from snuba.datasets import TimeSeriesDataset
+from snuba.datasets.dataset_schemas import DatasetSchemas
+from snuba.datasets.factory import get_dataset
+from snuba.datasets.schemas.join import (
+    JoinConditionExpression,
+    JoinCondition,
+    JoinedSchema,
+    JoinStructure,
+    JoinType,
+    SchemaJoinedSource,
+)
+from snuba.query.extensions import (
+    PerformanceExtension,
+    ProjectExtension,
+    QueryExtension,
+)
+from snuba.query.query import AliasedColumn
+from snuba.query.timeseries import TimeSeriesExtension
+
+
+class Groups(TimeSeriesDataset):
```

Comment:
Naming keeps being really really hard.
The idea behind this name is that it joins groups and assignees with events. So generally we start with groups.
Overall it will evolve to the whole datamodel, but at this point I don;t want to call it in a more generic way.